### DISCUSSION TITLE
AdonisJS View Model

### DISCUSSION THREAD

**Discussion by kafwe** [adonisjs/core#2187] [Help]
Is there a way to have viewModels (a model class that can hold only those properties that is required for a view) in AdonisJS app like in ASP.NET? If not, any alternative solutions would be greatly appreciated

**Comment by McSneaky** [comment]
No idea how they are in ASP.NET, but you can create your own models (classes) and put them in app/ViewModels (for example)

And then hold data in there and maybe deal with rendering views etc, so they just return rendered strings, which you can return in controller
